src/fileimpl.cpp

```diff
@@ -676,6 +677,16 @@ class Grouping
     return true;
   }
 
+  bool FileImpl::checkClusters() {
+    const cluster_index_type clusterCount = getCountClusters().v;
+    for ( cluster_index_type i = 0; i < clusterCount; ++i )
+    {
+      // For a read of each clusters (which will throw ZimFileFormatError in case of error)
+      readCluster(cluster_index_t(i));
```

Comment:
Other checks work by printing errors to `stderr` and returning `false` rather than by throwing an exception.

Reply:
Updated in fixup commit.